sdk/core/azure-core-serializer-avro-apache/src/main/java/com/azure/core/serializer/avro/apache/ApacheAvroSerializer.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.avro.apache;
+
+import com.azure.core.serializer.AvroSerializer;
+import com.azure.core.util.logging.ClientLogger;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.EncoderFactory;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Objects;
+
+/**
+ * Apache based implementation of the {@link AvroSerializer} interface.
+ */
+public class ApacheAvroSerializer implements AvroSerializer {
+    private static final Schema.Parser PARSER = new Schema.Parser();
+    private static final DecoderFactory DECODER_FACTORY = DecoderFactory.get();
+    private static final EncoderFactory ENCODER_FACTORY = EncoderFactory.get();
```

Comment:
Consider making encoder and decoder configurable too instead of using the default settings.

Reply:
Agreed, will move this over to using a builder pattern where this is settable.